ufl/algorithms/analysis.py

```diff
@@ -226,20 +231,36 @@ def extract_arguments_and_coefficients(a):
     if len(fcounts) != len(set(fcounts.values())):
         raise ValueError(
             "Found different coefficients with same counts.\n"
-            "The arguments found are:\n" + "\n".join(f"  {c}" for c in coefficients)
+            "The Coefficients found are:\n" + "\n".join(f"  {c}" for c in coefficients)
+        )
+
+    # Build count: instance mappings, should be one to one
+    gqcounts = {}
+    for gq in geometric_quantities:
+        if not isinstance(gq._domain, Mesh):
+            raise TypeError(f"{gq}._domain must be a Mesh: got {gq._domain}")
```

Comment:
What else com `gq._domain` be? is this relying on the deprecated definition that a `domain` can be defined just through a `ufl.Cell`?

Reply:
No. It checks if `gq._domain` is `Mesh` and not `MixedMesh`.